guacamole/src/main/webapp/app/navigation/directives/guacUserMenu.js

```diff
@@ -134,6 +135,64 @@ angular.module('navigation').directive('guacUserMenu', [function guacUserMenu()
                 return authenticationService.isAnonymous();
             };
 
+            /**
+             * All available actions for the current user.
+             */
+            $scope.actions = [ ];
+
+            // Initialize fullscreen functions
+            var docElem = document.documentElement;
+            var requestFullscreen = docElem.mozRequestFullScreen
+                || docElem.webkitRequestFullscreen
+                || docElem.msRequestFullscreen;
+            var exitFullscreen = document.mozCancelFullScreen
+                || document.webkitExitFullscreen
+                || document.msExitFullscreen;
+            if (!!requestFullscreen && !!exitFullscreen) {
+                // Bind browser-specific fullscreen functions
+                if (!docElem.requestFullscreen) {
+                    docElem.requestFullscreen = (requestFullscreen).bind(docElem);
+                }
+                if (!document.exitFullscreen) {
+                    document.exitFullscreen = (exitFullscreen).bind(document);
+                }
+    
+                /**
+                 * Toggles fullscreen for the Guacamole page.
+                 */
+                $scope.fullscreen = function fullscreen() {
+                    var fullscreenElem = !!(document.fullscreenElement 
+                        || document.msFullscreenElement
+                        || document.mozFullScreenElement
+                        || document.webkitFullscreenElement);
+    
+                    if (!fullscreenElem) {
+                        document.documentElement.requestFullscreen();
+                    }
+                    else {
+                        document.exitFullscreen();
+                    }
+                };
+    
+                /**
+                 * Action which requests fullscreen for the Guacamole page.
+                 */
+                var FULLSCREEN_ACTION = {
+                    name      : 'USER_MENU.ACTION_FULLSCREEN',
+                    className : 'fullscreen',
+                    callback  : $scope.fullscreen
+                };
+    
+                /**
+                 * Add fullscreen action for the current user.
+                 */
+                $scope.actions = $scope.actions.concat( FULLSCREEN_ACTION );
```

Comment:
Running this only within `if (!!requestFullscreen && !!exitFullscreen) { ... }` will result in the fullscreen option being available only if there is a vendor-prefixed version of those functions available. If a browser implements the standard but no vendor-prefixed version, the option will not be present.

Reply:
You're right. I should have added the standard version in `requestFullscreen` and `exitFullscreen`.
Thanks.